storm-core/src/clj/org/apache/storm/daemon/executor.clj

```diff
@@ -698,7 +699,7 @@
         executor-stats (:stats executor-data)
         {:keys [storm-conf component-id worker-context transfer-fn report-error sampler
                 open-or-prepare-was-called?]} executor-data
-        rand (Random. (Utils/secureRandomLong))
+        rand (XORShiftRandom. (Utils/secureRandomLong))
```

Comment:
Same reason as with the spout, but differently.  There is a much higher likelihood that this will be used in a non-thread safe way.  But it is less likely that it will result in a tuple being counted improperly, unless multiple threads are emitting tuples for the same tuple tree.  It is unlikely, but still not necessarily safe.